generate/go_client.tpl

```diff
@@ -43,9 +47,36 @@ func New(config *DialConfig) (Client, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	c := &client{
-		config: config,
+		config:       config,
+		interceptors: []connect.Interceptor{},
```

Comment:
Since `client` contains `config` and `config` contains `Interceptors`, why is it necessary to add an `interceptors` field to `client`? Couldn't you just add the below interceptors to `client.config`?

Reply:
I don't get it :see_no_evil: How does my suggestion allow this if if wasn't before? Can you give an example?

Reply:
Yes but `client.config` is lowercase so how would you change `client.config.Interceptors`? I think I'm missing something. We can discuss this in the next review session.